tests/positive/dynamic_rendering.cpp

```diff
@@ -998,3 +998,153 @@ TEST_F(PositiveDynamicRendering, WithShaderTileImageAndBarrier) {
     vk::CmdEndRendering(m_commandBuffer->handle());
     m_commandBuffer->end();
 }
+
+TEST_F(PositiveDynamicRendering, MatchingAttachmentFormats) {
+    TEST_DESCRIPTION(
+        "Draw with Dynamic Rendering with attachment specified as VK_NULL_HANDLE in VkRenderingInfoKHR, and with corresponding "
+        "format in VkPipelineRenderingCreateInfoKHR set to VK_FORMAT_UNDEFINED");
+
+    SetTargetApiVersion(VK_API_VERSION_1_1);
+
+    AddRequiredExtensions(VK_KHR_DYNAMIC_RENDERING_EXTENSION_NAME);
+
+    ASSERT_NO_FATAL_FAILURE(InitFramework());
+
+    if (!AreRequiredExtensionsEnabled()) {
+        GTEST_SKIP() << RequiredExtensionsNotSupported() << " not supported";
+    }
+
+    if (DeviceValidationVersion() < VK_API_VERSION_1_1) {
+        GTEST_SKIP() << "At least Vulkan version 1.1 is required";
+    }
+
+    auto dynamic_rendering_features = LvlInitStruct<VkPhysicalDeviceDynamicRenderingFeaturesKHR>();
+    auto features2 = GetPhysicalDeviceFeatures2(dynamic_rendering_features);
+    if (!dynamic_rendering_features.dynamicRendering) {
+        GTEST_SKIP() << "Test requires (unsupported) dynamicRendering";
+    }
+
+    ASSERT_NO_FATAL_FAILURE(InitState(nullptr, &features2));
+
+    const VkViewport viewport = {0, 0, 16, 16, 0, 1};
+    const VkRect2D scissor = {{0, 0}, {16, 16}};
+    m_viewports.push_back(viewport);
+    m_scissors.push_back(scissor);
+
+    VkShaderObj vs(this, bindStateVertShaderText, VK_SHADER_STAGE_VERTEX_BIT);
+    VkShaderObj fs(this, bindStateFragColorOutputText, VK_SHADER_STAGE_FRAGMENT_BIT);
+
+    VkDescriptorSetLayoutBinding dslb = {0, VK_DESCRIPTOR_TYPE_INPUT_ATTACHMENT, 1, VK_SHADER_STAGE_FRAGMENT_BIT, nullptr};
+    const VkDescriptorSetLayoutObj dsl(m_device, {dslb});
+    const VkPipelineLayoutObj pl(m_device, {&dsl});
+
+    auto pipeline_rendering_info = LvlInitStruct<VkPipelineRenderingCreateInfoKHR>();
+
+    VkPipelineObj pipeline_color(m_device);
+    pipeline_color.AddShader(&vs);
+    pipeline_color.AddShader(&fs);
+    pipeline_color.AddDefaultColorAttachment();
+    pipeline_color.SetViewport(m_viewports);
+    pipeline_color.SetScissor(m_scissors);
+
+    VkFormat color_formats[] = {VK_FORMAT_UNDEFINED};
+    pipeline_rendering_info.colorAttachmentCount = 1;
+    pipeline_rendering_info.pColorAttachmentFormats = color_formats;
+
+    auto create_info1 = LvlInitStruct<VkGraphicsPipelineCreateInfo>();
```

Comment:
```suggestion
    auto create_info_color = LvlInitStruct<VkGraphicsPipelineCreateInfo>();
```

avoid numbering please, always annoying when tests are edited in future and ordering is not kept